plugins/check_postgresql_replication

```diff
@@ -0,0 +1,139 @@
+                             ' (checked on master)')
+    parser.add_argument('-c', '--critical', type=int,
+                        help='Critical level for offset between master and slave'
+                             ' (checked on master)')
+    args = parser.parse_args()
+
+    if args.critical is not None and (args.warning is None or args.warning > args.critical):
+        args.warning = args.critical
+
+    return args
+
+
+def check_replication(args, cur):
+    try:
+        cur.execute("""
+            SELECT
+                client_addr,
+                pg_xlog_location_diff(pg_stat_replication.sent_location,
+                    pg_stat_replication.replay_location) AS pg_xlog_location_diff
+            FROM pg_stat_replication
+            WHERE state = 'streaming'""")
+    except:
+        print "Critical: Unable to SELECT FROM pg_stat_replication()"
+        return STATE['Critical']
+
+    replicas = cur.fetchall()
+    if len(replicas) == 0:
+        print "Critical: No Streaming Replica's are attached"
+        return STATE['Critical']
+
+    if len(replicas) > 1:
+        print "Unknown: There are more then 1 Streaming Replica's attached. This is unsupported."
+        return STATE['Unknown']
+
+    replica = replicas[0]
+
+    state = 'Ok'
+    if args.critical is None:
+        state = 'Unknown'
+    elif replica['pg_xlog_location_diff'] > args.critical:
+        state = 'Critical'
+    elif replica['pg_xlog_location_diff'] > args.warning:
+        state = 'Warning'
+
+    print '%s: Slave %s is %s bytes behind' % (
+        state,
+        replica['client_addr'],
+        replica['pg_xlog_location_diff']
+    )
+
+    if args.critical is None:
+        print "Unknown limit: argument --critical is missing"
+
+    return STATE[state]
+
+
+def check_slave(args, cur):
+    try:
+        cur.execute("""SELECT setting FROM pg_settings WHERE name='data_directory'""")
+    except:
+        print "Unknown: Unable to SELECT data_directory FROM pg_settings"
+        return STATE['Unknown']
+    data_directory = cur.fetchone()[0]
+
+    try:
+        cur.execute("""SELECT setting::int FROM pg_settings WHERE name='archive_timeout'""")
+    except:
+        print "Unknown: Unable to SELECT archive_timeout FROM pg_settings"
+        return STATE['Unknown']
+    archive_timeout = cur.fetchone()[0]
+
+    donefiles = glob.glob('%s/*.done' % os.path.join(data_directory, 'pg_xlog/archive_status'))
+    if not donefiles:
+        print "Unknown: No .done files found in pg_xlog/archive_status"
+        return STATE['Unknown']
+    last_donefile = max(donefiles, key=os.path.getmtime)
+    donefile_basename = os.path.basename(last_donefile)
+    donefile_mtime = int(os.path.getmtime(last_donefile))
+    donefile_seconds_ago = int(time.time()) - donefile_mtime
+
+    state = 'Ok'
+    if donefile_seconds_ago > archive_timeout*3:
+        state = 'Critical'
+    if donefile_seconds_ago > archive_timeout*2:
```

Comment:
The ```warning``` check should either be done before the ```critical``` check, or you might want to use elseif instead, to prevent the state from falling through.